### PR TITLE
feat(backend): Redirect after login

### PR DESCRIPTION
**Pull request by PAtB1992**

### CODE REVIEW

File: packages/auth-magic-link-nest/src/magic-link/callbackUrl.middleware.ts
Comment (cape-dev) [inline packages/auth-magic-link-nest/src/magic-link/callbackUrl.middleware.ts:1]: please use kebab case for filenames

File: packages/auth-magic-link-nest/src/magic-link/callbackUrl.middleware.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Injectable, NestMiddleware } from '@nestjs/common';
+import { NextFunction, Request, Response } from 'express';
+
+@Injectable()
+export class callbackUrlMiddleware implements NestMiddleware {
+  use(req: Request, res: Response, next: NextFunction): any {
+    const callbackUrl = req.query.callbackUrl;
+    if (!callbackUrl) {
+      req.query.callbackUrl = 'default';
```
Comment (cape-dev) [inline packages/auth-magic-link-nest/src/magic-link/callbackUrl.middleware.ts]: there should be something more sophisticated than just 'default'. It should be something that cannot be the callbackUrl by normal usage of this plugin.

additionally this string should be extracted as a constant. we have a file called `constants.ts` where such values can be placed.

File: packages/auth-magic-link-nest/src/magic-link/export.types.ts [hunk 2]
```diff
@@ -25,7 +29,7 @@ export interface AuthMagicLinkUtil<DBUser = any> {
     refreshTokenHash: string,
   ) => void | Promise<void>;
   sendMagicLink: (
-    user: DBUser,
+    user: MagicLinkUser,
```
Comment (cape-dev) [inline packages/auth-magic-link-nest/src/magic-link/export.types.ts]: basically we have the extracted query params from the API call here which we stated in `magic-link-strategy.ts` in the `userFields` array. I think we should rename the parameter in this function to something like `extractedQueryParams`. Also the interface `MagicLinkUser` should be renamed to something similar.

File: packages/auth-magic-link-nest/src/magic-link/login.controller.ts [hunk 3]
```diff
@@ -29,17 +36,23 @@ export class LoginController {
   async validate(
     @Res({ passthrough: true }) response: Response,
     @ExtractEmailFromMagicLink() email: string,
+    @Query('callbackUrl') callbackUrl: string,
   ) {
     const { frontendUrls } = this.config;
 
     const { userId } = await this.util.findOrCreateByEmail({ email });
+    console.log(callbackUrl);
```
Comment (cape-dev) [inline packages/auth-magic-link-nest/src/magic-link/login.controller.ts]: remove console log

File: packages/auth-magic-link-nest/src/magic-link/login.controller.ts [hunk 4]
```diff
@@ -29,17 +36,23 @@ export class LoginController {
...
 
     await this.cookieTokenService.prepareResponseWithNewTokens(
       response,
       userId,
     );
 
-    response.redirect(frontendUrls.auth.redirect.success);
+    if (callbackUrl !== 'default') {
```
Comment (cape-dev) [inline packages/auth-magic-link-nest/src/magic-link/login.controller.ts]: use the constant for the condition

Comment (cape-dev) [reply packages/auth-magic-link-nest/src/magic-link/login.controller.ts]: in fact... we should not set the `callbackUrl` if we have the "default" value which in turn leads to the situation where we only have to validate if the `callbackUrl` is present at this stage or not

File: packages/auth-magic-link-nest/src/magic-link/magic-link.strategy.ts [hunk 5]
```diff
@@ -34,7 +33,7 @@ export class MagicLinkStrategy extends PassportStrategy(Strategy) {
                 this.config.paths.proxy ?? '',
                 this.config.paths.magicLink.validate,
               ])
-              .query({ token })
+              .query({ token, callbackUrl: user.callbackUrl })
```
Comment (cape-dev) [inline packages/auth-magic-link-nest/src/magic-link/magic-link.strategy.ts]: as mentioned above, we should not include the callbackUrl as a query parameter in the generated magic link if it has the "default" value:

```
...
import { DEFAULT_CALLBACK_URL } = '.../contants.ts';

.query({ 
    token,  
    ...(user.callbackUrl && user.callbackUrl !== DEFAULT_CALLBACK_URL) && { callbackUrl: user.callbackUrl }
})
```

**Review by cape-dev — COMMENTED** [review]
see comments

File: packages/auth-magic-link-nest/src/magic-link/callbackUrl.middleware.ts (hunk 1 above)
Comment (PAtB1992) [reply packages/auth-magic-link-nest/src/magic-link/callbackUrl.middleware.ts]: added DefaultCallbackUrl as a constant in constants.ts

**Pull request merged**